File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/oauth/OAuthIntentService.java
package kore.botssdk.audiocodes.oauth;

import android.content.Context;
import android.content.Intent;
import android.util.Log;

import androidx.core.app.JobIntentService;

public class OAuthIntentService extends JobIntentService {

    private static final int JOB_ID = 1000;

    public static void enqueueWork(Context context, Intent work) {
        enqueueWork(context, OAuthIntentService.class, JOB_ID, work);
    }

    @Override
    public void onHandleWork(Intent intent) {
        String id = intent.getData().toString();
        Log.d(OAuthManager.TAG, "onHandleWork: " + id);
        if(id.equals(OAuthReceiver.Type.ACCESS_TOKEN.toString())){
            Log.d(OAuthManager.TAG, "onHandleWork authorize");
           // OAuthManager.getInstance().setAccessTokenAlarm();
        }
        else if(id.equals(OAuthReceiver.Type.REFRESH_TOKEN.toString())){
            Log.d(OAuthManager.TAG, "onHandleWork refresh");
            OAuthManager.getInstance().refreshToke();
        }
    }
}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/oauth/OAuthIntentService.kt
package com.kore.ui.audiocodes.oauth

import android.content.Context
import android.content.Intent
import androidx.core.app.JobIntentService
import com.kore.common.utils.LogUtils

class OAuthIntentService : JobIntentService() {
    private val oAuthManager: OAuthManager = OAuthManager.getInstance()
    public override fun onHandleWork(intent: Intent) {
        val id = intent.data.toString()
        LogUtils.d(OAuthManager.TAG, "onHandleWork: $id")
        if (id == OAuthReceiver.Type.ACCESS_TOKEN.toString()) {
            LogUtils.d(OAuthManager.TAG, "onHandleWork authorize")
            // OAuthManager.getInstance().setAccessTokenAlarm();
        } else if (id == OAuthReceiver.Type.REFRESH_TOKEN.toString()) {
            LogUtils.d(OAuthManager.TAG, "onHandleWork refresh")
            oAuthManager.refreshToken()
        }
    }

    companion object {
        private const val JOB_ID = 1000

        @JvmStatic
        fun enqueueWork(context: Context, work: Intent) {
            enqueueWork(context, OAuthIntentService::class.java, JOB_ID, work)
        }
    }
}